File: chapter 1/ex_1_7.js
const average = (x, y) => (x + y) / 2;
const improve = (guess, x) => average(guess, (x / guess));
const smallEnough = x => x / 100;
const goodEnough = (guess, x) => Math.abs((guess ** 2) - x) < smallEnough(x);

const sqrtIter = (guess, x) => {
  return goodEnough(guess, x) ? guess : sqrtIter(improve(guess, x), x);
};

console.log(sqrtIter(1, 9));

File: chapter 1/ex_1_7.erl
-module('ex_1_7').
-export([sqrt_iter/2]).
-compile([debug_info]).

average(X, Y) ->
    (X + Y) / 2.

improve(Guess, X) ->
    average(Guess, (X / Guess)).

small_enough(X) ->
    X / 100.

good_enough(Guess, X) ->
    abs(math:pow(Guess, 2) - X) < small_enough(X).

sqrt_iter(Guess, X) ->
    case good_enough(Guess, X) of
	true -> io:format("Result: ~p ~n", [Guess]);
	false -> sqrt_iter((improve(Guess, X)), X)
    end.
